pandas/core/generic.py

```diff
@@ -7680,8 +7693,42 @@ def between_time(
         if not isinstance(index, DatetimeIndex):
             raise TypeError("Index must be DatetimeIndex")
 
+        # If user has passed any of the depreciated arguments
+        # ('include_start', 'include_end') AND the new argument ('inclusive')
+        if (include_start != lib.no_default or include_end != lib.no_default) and (
+            inclusive is not None
+        ):
+            raise ValueError(
+                "Depreciated arguments `include_start` and `include_end`"
+                "cannot be passed if `inclusive` has been given."
+            )
+        # If any of the depreciated arguments ('include_start', 'include_end')
+        # have been passed
+        elif (include_start != lib.no_default) or (include_end != lib.no_default):
+            warnings.warn(
+                "`include_start` and `include_end` are deprecated in"
+                "favour of `inclusive`.",
+                FutureWarning,
+                stacklevel=2,
+            )
+            if include_start == lib.no_default:
+                include_start = True
+            if include_end == lib.no_default:
+                include_end = True
+        else:  # On arg removal inclusive can default to "both"
+            if inclusive is None:
+                inclusive = "both"
+            elif inclusive not in ["both", "neither", "left", "right"]:
+                raise ValueError(
+                    f"Inclusive has to be either string of 'both',"
```

Comment:
unnecessary `f-string` on this line

Reply:
I had included it for consistency as mentioned [here](inclusive: str | None = None). A bit confused now, should I remove it?

Reply:
Actually, don't worry about it, if there's no automated rule about this yet then it's not worth changing this, sorry for this comment